wandb/sdk/launch/_project_spec.py

```diff
@@ -316,7 +328,11 @@ def docker_image(self) -> Optional[str]:
         Returns:
             Optional[str]: The Docker image or None if not specified.
         """
-        return self._docker_image
+        if self._docker_image:
+            return self._docker_image
+        if self.job_base_image:
+            return self.job_base_image
```

Comment:
It feels strange to conflate this, I think this should point to the unmodified image containing code, why do we do this?

Reply:
this was a bit of a hack so that we wouldn't have to modify some bits of code that use the runner and set the `image_uri` argument based on `docker_image`. I've removed it and updated the upstream prs